src/sentry_plugins/slack/plugin.py

```diff
@@ -239,18 +234,26 @@ def notify(self, notification, raise_exception=False):
                 }
             ]
         }
+        client = self.get_client(project)
 
-        if username:
-            payload["username"] = username.encode("utf-8")
+        if client.username:
+            payload["username"] = client.username.encode("utf-8")
 
-        if channel:
-            payload["channel"] = channel
+        if client.channel:
+            payload["channel"] = client.channel
 
-        if icon_url:
-            payload["icon_url"] = icon_url
+        if client.icon_url:
+            payload["icon_url"] = client.icon_url
 
         values = {"payload": json.dumps(payload)}
+        client.request(values)
 
+    def get_client(self, project):
+        webhook = self.get_option("webhook", project)
         # Apparently we've stored some bad data from before we used `URLField`.
         webhook = webhook.strip(" ")
```

Comment:
nit: I think you could maybe put `.strip()` on the same line and remove the `" "`